### PR TITLE
Spring boot security implementation

### PR DESCRIPTION
**Pull request by ValentynSharshon**

### CODE REVIEW

**Review by JohnGrey17 — APPROVED** [review]
Little bit hard to review because lot of separate commit but anyways code looks correct

File: src/main/java/com/gmail/woosay333/onlinebookstore/exception/handler/ControllerExceptionHandler.java [hunk 1]
```diff
@@ -49,6 +54,21 @@ protected ResponseEntity<Object> handleHttpMessageNotReadable(
...
+    @ExceptionHandler(AuthenticationException.class)
+    protected ResponseEntity<Object> handleAuthenticationException(AuthenticationException ex) {
+        return getResponseEntity(UNAUTHORIZED, ex.getMessage());
+    }
+
+    @ExceptionHandler(UsernameNotFoundException.class)
+    protected ResponseEntity<Object> handleUsernameNotFoundException(UsernameNotFoundException ex) {
+        return getResponseEntity(NOT_FOUND, ex.getMessage());
+    }
+
```
Comment (andrii-hoienko) [inline src/main/java/com/gmail/woosay333/onlinebookstore/exception/handler/ControllerExceptionHandler.java]: These exceptions handled automatically as far as I remember

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Good job! Left one question

File: src/main/java/com/gmail/woosay333/onlinebookstore/exception/handler/ControllerExceptionHandler.java (hunk 1 above)
Comment (ValentynSharshon) [reply src/main/java/com/gmail/woosay333/onlinebookstore/exception/handler/ControllerExceptionHandler.java]: You're probably right. Found a topic about it on [stackoverflow](https://stackoverflow.com/questions/59417122/how-to-handle-usernamenotfoundexception-spring-security).
I'll fix it now and throw it in again for checking, thanks.

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**